core/src/executor.rs

```diff
@@ -318,20 +320,18 @@ impl<'de> DeserializeSeed<'de> for WasmSeed<'_, LoadedExecutor> {
                 M: MapAccess<'de>,
             {
                 while let Some(key) = map.next_key::<String>()? {
-                    if key.as_str() == "module" {
-                        let wrapper: Base64Wrapper = map.next_value()?;
-                        let module = wasm::deserialize_module(self.loader.engine, wrapper.0)
-                            .map_err(serde::de::Error::custom)?;
-                        return Ok(LoadedExecutor::new(module));
+                    if key.as_str() == "raw_executor" {
```

Comment:
I think you can somewhat simplify the `Deserialization` code by using serde to define what struct you expect instead of matching on it manually (and no need for visitors!):

```rust
// a copy of `LoadedExecutor` without the `module` field
#[derive(Deserialize)]
struct LoadedExecutor {
    raw_executor: data_model_executor::Executor,
}

let executor = LoadedExecutor::deserialize(deserializer)?;

self::LoadedExecutor::load(self.engine, executor.raw_executor)
    .map_err(serde::de::Error::custom)
```